base/tks/src/main/java/org/dogtagpki/server/tks/servlet/TokenServlet.java

```diff
@@ -2815,10 +2883,28 @@ private void processComputeSessionKeysSCP03(HttpServletRequest req, HttpServletR
         }
 
         if (serversideKeygen == true) {
-            if (serverSideValues.size() == 3) {
+            logger.debug(method + " serversideValues.size: " + serverSideValues.size());
+	    for(int i = 0 ; i < serverSideValues.size(); i++) {
+                logger.debug(method + " val:" + i  + " : " + serverSideValues.get(i));
+            }
+
+            if (serverSideValues.size() >= 3) {
+                logger.debug(method + " size >= 3 ");
                 drm_trans_wrapped_desKeyString = serverSideValues.get(2);
                 kek_wrapped_desKeyString = serverSideValues.get(0);
                 keycheck_s = serverSideValues.get(1);
+                keycheck_aes_s = serverSideValues.get(2);
+
+                if(serverSideValues.size() >= 5) {
+			logger.debug(method + " size >= 5");
+                    drm_trans_wrapped_aesKeyString = serverSideValues.get(3);
+                }
+
+		if(serverSideValues.size() >= 6) {
+			logger.debug(method + " size >= 6");
+                    kek_wrapped_aesKeyString = serverSideValues.get(5);
```

Comment:
could you explain in a comment at the top of the "if (serversideKeygen == true) {" stetement what value cases are there for serverSideValues?  Also, seeing that this method is processComputeSessionKeysSCP03, should it be assumed that you are mainly dealing with aes keys?